pkg/logs/input/file/tailer.go

```diff
@@ -58,9 +59,12 @@ type Tailer struct {
 func NewTailer(outputChan chan *message.Message, source *config.LogSource, path string, sleepDuration time.Duration, isWildcardPath bool) *Tailer {
 	// TODO: remove those checks and add to source a reference to a tagProvider and a lineParser.
 	var parser lineParser.Parser
-	if source.GetSourceType() == config.KubernetesSourceType {
+	switch source.GetSourceType() {
+	case config.KubernetesSourceType:
 		parser = kubernetes.Parser
-	} else {
+	case config.DockerSourceType:
```

Comment:
Could you add some tests related to the issue we discussed earlier: 
* logs splited by docker when oversized. so 2 independent logs are passed to decoder the later one without '\n' in the message. 
* decoder with multilinehandler to see the behaviour of this parser.

Reply:
I don't see the tests for the 1st scenario